bedwars-plugin/src/main/java/com/andrei1058/bedwars/arena/Arena.java

```diff
@@ -2536,6 +2539,10 @@ public boolean startReSpawnSession(Player player, int seconds) {
                 PaperSupport.teleportC(player, getReSpawnLocation(), PlayerTeleportEvent.TeleportCause.PLUGIN);
                 player.setAllowFlight(true);
                 player.setFlying(true);
+                player.setHealth(20);
+                player.getActivePotionEffects().forEach(effect -> player.removePotionEffect(effect.getType()));
```

Comment:
What is the point of removing potion effects? Wont this just cause issues with addons applying a potion effect to players?

Reply:
cuz if the player dies having jump boot or invis effect, the effect will continue while respawning

Reply:
i guess i should rather add that to checkPlayerHealth method before calling PlayerDeathEvent so there wont be any issues if another addon tries to apply a potion effect

Reply:
on the other side, if a PlayerDeathEvent is thrown an addon could also catch this and reapply any potion effects on respawn.

Reply:
yeah, I've now made it to remove potion effects before calling the PlayerDeathEvent, i guess thats better.